sound/soc/sof/intel/hda.c

```diff
@@ -1063,12 +1081,32 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 			 */
 			unique_id = SDW_UNIQUE_ID(adr);
 			if (link->num_adr == 1 ||
-			    ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID ||
 			    ids[j].id.unique_id == unique_id) {
 				dev_dbg(bus->dev,
 					"found %x at link %d\n",
 					part_id, link_id);
 				break;
+			} else if (ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID) {
+				unsigned int part_id2;
+
+				/* make sure there is no the same codec on the link */
+				for (k = 0; k < link->num_adr; k++) {
+					if (k == i)
+						continue;
+					part_id2 = SDW_PART_ID(link->adr_d[k].adr);
+					if (part_id2 == part_id) {
+						dev_dbg(bus->dev,
+							"wrong machine descriptor: link %d has duplicate part_id %d already tagged as unique\n",
+							link_id, part_id);
+						return false;
+					}
+				}
+				if (k == link->num_adr) {
```

Comment:
when k != link->num_adr ?  if (part_id2 == part_id) and it will return false. so do we need this check ?

Reply:
`if (k == link->num_adr)` is true when we checked all adr and found no the same codec in the link. So we can say we found the codec.

Reply:
I mean just removing if (k == link->num_adr) since it is definitely true.

Reply:
> I mean just removing if (k == link->num_adr) since it is definitely true.

Thanks @RanderWang Update now.